core/templates/components/forms/custom-forms-directives/thumbnail-uploader.directive.ts

```diff
@@ -198,11 +198,7 @@ angular.module('oppia').directive('thumbnailUploader', [
               $scope.thumbnailIsLoading = true;
               var filename = ImageUploadHelperService.generateImageFilename(
                 data.dimensions.height, data.dimensions.width, 'svg');
-              ImageLocalStorageService.saveImage(
-                filename, data.newThumbnailDataUrl);
-              ImageLocalStorageService.setImageBgColor(data.newBgColor);
               $scope.newThumbnailDataUrl = data.newThumbnailDataUrl;
-              ContextService.setImageSaveDestinationToLocalStorage();
```

Comment:
This should be handled by the page using this directive, since only it can reset the context after the use is done. So removed it from here and added it to TopicCreationService.